### PR TITLE
322 Authenticate oid request

### PR DESCRIPTION
**Pull request by dylansalay**
Protects OID Minter Service using HTTP Basic Authentication

Screen Shot 2020-07-10 at 3 38 00 PM

### CODE REVIEW

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+  # ==> Controller configuration
+  # Configure the parent class to the devise controllers.
+  # config.parent_controller = 'DeviseController'
+
+  # ==> Mailer Configuration
+  # Configure the e-mail address which will be shown in Devise::Mailer,
+  # note that it will be overwritten if you use your own mailer class
+  # with default "from" parameter.
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
```
Comment (maxkadel) [inline config/initializers/devise.rb:27]: if we have this un-commented, should we make this a real email address? or should we comment it out?

File: db/migrate/20200709205615_devise_create_users.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      ## Lockable
+      # t.integer  :failed_attempts, default: 0, null: false # Only if lock strategy is :failed_attempts
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
+
+
+      t.timestamps null: false
+    end
+
+    add_index :users, :email,                unique: true
```
Comment (maxkadel) [inline db/migrate/20200709205615_devise_create_users.rb:39]: I know this is probably just all the defaults, but do we really want to use email? or do we want to use username? I think if we do want to use username it'd be fine to do it on another pass.

File: spec/requests/oid_minter_request_spec.rb [hunk 3]
```diff
@@ -2,69 +2,93 @@
...
-      expect(json_response['oids']).not_to be_nil
-      oids = json_response['oids']
-      expect(oids).to be_a Array
-      expect(oids.length).to be 1
-      expect(oids[0]).to be_a Integer
+  let(:user) { FactoryBot.create(:user) }
+  describe "when authenticated" do
+    context "Requesting a single OID" do
+      it "returns a successful JSON response" do
+        headers = login_header(user).merge('ACCEPT' => "application/json")
```
Comment (maxkadel) [inline spec/requests/oid_minter_request_spec.rb:9]: I wasn't familiar with this `login_header(user)` method, that seems super handy.

**Review by bess — APPROVED** [review]

File: config/initializers/devise.rb (hunk 1 above)
Comment (dylansalay) [reply config/initializers/devise.rb:27]: I think it makes sense to comment it out for now. Maybe we add an email if we choose to implement devise authentication for the application?

File: db/migrate/20200709205615_devise_create_users.rb (hunk 2 above)
Comment (dylansalay) [reply db/migrate/20200709205615_devise_create_users.rb:39]: Yes, we just went with the defaults. But we can discuss and change to username if that is a better solution.

File: spec/requests/oid_minter_request_spec.rb (hunk 3 above)
Comment (dylansalay) [reply spec/requests/oid_minter_request_spec.rb:9]: It's not a Rails built in helper method - just a custom method we created in spec/support/auth_helper.rb to handle the authentication login.

**Pull request merged**